### PR TITLE
Added new operation matchers

### PR DESCRIPTION
**Pull request by yogeshjain999**
- Removed existing helpers `use_model` and `present_model` matchers (So not backward compatible).
- Added few new matchers like `be_success`, `be_failure`, `have_properties` and `have_invalid_properties` :beers: 
- Added ability to define default values to be passed to an operation using `default_params` and ability to override them using `with` helper.

### CODE REVIEW

**Comment by fernandes** [comment]
Usually is a good practice (specially for from users understand the
releases/history) make all modifications and then one single commit to bump
version :)

**Review by emaglio — CHANGES_REQUESTED** [review]
I agree with @fernandes, better to have a version bump in another PR. We follow this wiki for this kind of stuff: https://github.com/trailblazer/trailblazer/wiki/Maintenance
Anyway before releasing something let's have more stuff in it and then we can release a 1.x version directly. I think we could then add compatibility for activities releasing a 2.x so we are aligned with TRB 2.x.

Apart of that look good :+1: - just a small comment about `Contract::DSL`

**Comment by yogeshjain999** [comment]
Thanks for the review @fernandes @emaglio  I've updated the commits as per that.

@emaglio , agreed that we should add more support before releasing. I need to some tests too :sweat_smile: 
I've created PR to master as of now, if you or anyone with write access can create 0.3.0 branch, I'll change it to point to that.

**Comment by emaglio** [comment]
@yogeshjain999 good point about the tests - let's add the tests for these matches here then :+1:

File: README.md
Comment (emaglio) [inline README.md]: Sorry also this is deprecated - can you please use a "normal" contract instead to use this?

Comment (yogeshjain999) [reply README.md]: Sure, I'll update that. I liked this way when we've only few properties in form.

Comment (fernandes) [reply README.md]: @yogeshjain999 I do too hahah

the problem is to _resuse_ the form, need to `Thing::Contract::Create['contract.default.class']`, weird no? hahahah

File: lib/rspec/trailblazer/operation/matchers.rb
Comment (myabc) [inline lib/rspec/trailblazer/operation/matchers.rb]: One of RSpec's biggest selling points is that assertions read "like English". _be success_ doesn't sound read well. How about `be_successful` or `run_with_success` or `succeed`?

**Comment by myabc** [comment]
I have a few qualms about matcher naming.. Otherwise the implementation looks excellent.

File: lib/rspec/trailblazer/operation/matchers.rb
Comment (yogeshjain999) [reply lib/rspec/trailblazer/operation/matchers.rb]: I actually mapped `be_success` to `operation.success?` method, just like `be_valid` to `obj.valid?` or something like that. But I guess we could go with `be_successful` also.

File: README.md
Comment (yogeshjain999) [reply README.md]: Yeah right, haha

Comment (yogeshjain999) [reply README.md]: I've updated it @emaglio , thanks for pointing it out.

**Review by emaglio — APPROVED** [review]

**Comment by emaglio** [comment]
@yogeshjain999 can you please rename `be_success` with `be_successful` as per @myabc suggestion?
After that, this PR is ready to be merged :+1: - good job

**Comment by yogeshjain999** [comment]
Done @emaglio :beers:

**Pull request merged**
